src/test/scala/io/iohk/ethereum/db/cache/SimpleLRUSpec.scala

```diff
@@ -0,0 +1,53 @@
+package io.iohk.ethereum.db.cache
+
+import org.scalatest.wordspec.AnyWordSpec
+
+class SimpleLRUSpec extends AnyWordSpec {
+
+  var time = 0L
+
+  private object MockedLRU extends SimpleLRU[Int](10, 100) {
+    override protected def currentTime: Long = SimpleLRUSpec.this.time
+  }
+
+  "It" should {
+
+    "Respond false with all missing entries and preserve length" in {
+      val results = (0 until 100).map { i => MockedLRU.checkAndRefreshEntry(i) }
+      assert( results.forall( _ == false ) )
+    }
+
+    "Drop the records according to maxElements" in {
+      val existing = (90 until 100).map { i => MockedLRU.checkAndRefreshEntry(i) }
+      assert( existing.forall( _ == true ) )
+
+      // maxElements guaranteed that we have no space for those
+      val absent = (0 until 10).map { i => MockedLRU.checkAndRefreshEntry(i) }
```

Comment:
Why did the cache remove elements `0 until 10`? If elements `90 until 100` already existed there should still be space for those, right?

Reply:
This test makes sure that when `90 until 100` are inside - and storage limit is 10 there's definitely no space for 0 to 10 and they are absent